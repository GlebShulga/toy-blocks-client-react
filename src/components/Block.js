import React from "react";

import {
  Typography,
  makeStyles,
  Box,
} from "@material-ui/core";

const Block = ({ block }) => {

  const index = block?.attributes.index.toString().padStart(3, "0");
  const text =
    block?.attributes.data.charAt(0).toUpperCase() +
    block?.attributes.data.slice(1);

  const classes = useStyles();
  return (
    <Box className={classes.block}>
      <Typography variant="subtitle1" className={classes.index}>
        {index}
      </Typography>
      <Typography variant="subtitle1" className={classes.text}>
        {text}
      </Typography>
    </Box>
  );
};

const useStyles = makeStyles((theme) => ({
  block: {
    display: "flex",
    flexDirection: "column",
    backgroundColor: "#e0e0e0",
    margin: "0.2rem 0.2rem",
    padding: "0.2rem 0.5rem",
  },
  index: {
    color: "#1976d2",
    fontWeight: "bold",
    fontSize: theme.typography.pxToRem(11),
  },

  text: {
    fontSize: theme.typography.pxToRem(14),
    lineHeight: 2,
  },
}));


export default Block;
