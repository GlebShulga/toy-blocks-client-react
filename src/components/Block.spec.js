import React from "react";
import { shallow } from "enzyme";
import { create } from "react-test-renderer";
import Block from "./Block";

describe("<Block />", () => {
  const blocks = {
    data: [
      {
        id: "5",
        type: "blocks",
        attributes: {
          index: 1,
          timestamp: 1530679678,
          data: "The Human Torch",
          "previous-hash": "KsmmdGrKVDr43/OYlM/oFzr7oh6wHG+uM9UpRyIoVe8=",
          hash: "oHkxOJWOKy02vA9r4iRHVqTgqT+Afc6OYFcNYzyhGEc=",
        },
      },
      {
        id: "6",
        type: "blocks",
        attributes: {
          index: 2,
          timestamp: 1530679684,
          data: "is denied",
          "previous-hash": "oHkxOJWOKy02vA9r4iRHVqTgqT+Afc6OYFcNYzyhGEc=",
          hash: "9xr57PW8RVzeOniNP2lPVzAOu97x12mpDgjv70z5vo4=",
        },
      },
      {
        id: "7",
        type: "blocks",
        attributes: {
          index: 3,
          timestamp: 1530679689,
          data: "a bank loan",
          "previous-hash": "9xr57PW8RVzeOniNP2lPVzAOu97x12mpDgjv70z5vo4=",
          hash: "YGLfNDMC2x2m5kwb3q+Ne/uCL4sFUnX/sQwzuwijx8A=",
        },
      },
    ],
  };


  it("should contain .makeStyles-block-1 wrapper", () => {
    const component = shallow(<Block />);
    const wrapper = component.find(".makeStyles-block-1");
    expect(wrapper.length).toBe(1);
  });

  it("should match snapshot", () => {
    const component = create(<Block></Block>);
    const tree = component.toJSON();

    expect(tree).toMatchSnapshot();
  });

});
