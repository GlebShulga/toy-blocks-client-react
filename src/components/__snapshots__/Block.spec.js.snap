// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Block /> should match snapshot 1`] = `
<div
  className="MuiBox-root MuiBox-root-4 makeStyles-block-1"
>
  <h6
    className="MuiTypography-root makeStyles-index-2 MuiTypography-subtitle1"
  />
  <h6
    className="MuiTypography-root makeStyles-text-3 MuiTypography-subtitle1"
  >
    NaN
  </h6>
</div>
`;
